frontend/ui/widget/focusmanager.lua

```diff
@@ -369,7 +369,7 @@ end
 
 function FocusManager:_verticalStep(dy)
     local x = self.selected.x
-    if type(self.layout[self.selected.y + dy]) ~= "table" or self.layout[self.selected.y + dy] == {} then
```

Comment:
At a glance I think this is trying to say something like `#self.layout[self.selected.y + dy] == 0`, not `next()`.

Reply:
Depends on whether layout is an array or a hash (I... don't remember ;p).

Reply:
So, yeah, `next` is the safer bet if that's an unknown, as # won't work for a hash, while next will handle both, at least in that very specific context, since an empty table is arguably neither a hash nor an array ;p.

Reply:
You're not wrong, but you're not right either. ;-) The first one is already true if not table. But fair enough that next() might be a safer way to phrase it on general principle, even if I think it's an obscurantist way of phrasing it.

Reply:
`next()` is obscure and ugly - but it is indeed the only way to check if a hash table is empty. We use that in a few places.

Reply:
From the official documentation of Lua:
> In particular, you can use next(t) to check whether a table is empty.
<https://www.lua.org/manual/5.4/manual.html#6.1>

I had no idea about what this table was supposed to contain. The `#` operator would return 0 for `[nil, "a"]`. AFAIK, `next()` is the only safe way to check if a table is empty.

Reply:
> The # operator would return 0 for [nil, "a"].

And I'm saying this reads as if *that's exactly what's intended*.

Reply:
Anyway, this change is probably fine.